src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2381,6 +2381,11 @@ internal Binder GetBinder(CSharpSyntaxNode syntaxNode)
                     {
                         diagnostics.Add(ErrorCode.ERR_PartialMethodMustHaveLatent, method.Locations[0], method);
                     }
+
+                    if (method.IsPartialDefinition && (object)method.PartialImplementationPart == null)
+                    {
+                        method.DeclaringCompilation.SymbolDeclaredEvent(method);
```

Comment:
Yes, but that's OK. If a receiver of the event attempts to do anything with the member list, they will have to wait for it to be completed and stored. Since they are necessarily on a different thread, no problem.